.github/workflows/main.yml

```diff
@@ -0,0 +1,60 @@
+name: Lint
+
+on:
+    # Trigger the workflow on push or pull request,
+    # but only for some branches
+    push:
+        branches:
+            - master
+            - plugin
+    pull_request:
+        branches:
+            - master
+            - plugin
+
+jobs:
+    run-linters:
+        name: Run linters and TS checks
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Check out Git repository
+              uses: actions/checkout@v3
+
+            - name: Set up Node.js
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 16
+
+            # ESLint and Prettier must be in `package.json`
+            - name: Install Node.js dependencies
+              run: npm i
+
+            - name: Run TypeScript & file checks
+              run: |
+                set -e
```

Comment:
Ideally this would be multiple steps, not just one big step

Reply:
https://github.com/0xSpaceShard/starknet-hardhat-example/pull/122/commits/c51bef13f565d157a5c3cd7a759168ec426e67ba